routes/notes.js

```diff
@@ -42,79 +53,132 @@ router.get('/:id', (req, res, next) => {
     return next(err);
   }
 
+
+  //is this is a note id or a folder id?
+
+  //find note
   Note.findById(id)
+    .populate('folderId')
+    .populate('tags')
     .then(result => {
-      if (result) {
+      if (result){
         res.json(result);
-      } else {
+      } else { 
         next();
       }
     })
-    .catch(err => {
+
+    .catch( err => {
       next(err);
     });
+    
 });
+ 
 
+ 
 
 /* ========== POST/CREATE AN ITEM ========== */
 router.post('/', (req, res, next) => {
-  const { title, content } = req.body;
+
+  const { title, content, folderId, tags } = req.body;
+
+  const newNote = { title, content, folderId, tags };
+ 
 
   /***** Never trust users - validate input *****/
   if (!title) {
     const err = new Error('Missing `title` in request body');
     err.status = 400;
     return next(err);
+ 
   }
 
-  const newNote = { title, content };
+  //IS THIS RIGHT?
```

Comment:
> // IS THIS RIGHT?

No. `tags` is an array so you need to iterate over the array and check each value